tests/codegen/scratch_buffer.cpp

```diff
@@ -65,6 +65,20 @@ TEST(codegen, str_stack)
                                LARGE_ON_STACK_LIMIT);
 }
 
+TEST(codegen, map_value_scratch_buf)
```

Comment:
Maybe we should add some tests for some different types as they seem to determine if they get a buffer or not.

Reply:
Added tuple (with different sizes to trigger the !isSameSizeRecursive condition) and ints. Let me know if you'd like more types